derotation/analysis/incremental_derotation_pipeline.py

```diff
@@ -428,30 +464,259 @@ def register_rotated_images(
 
         return registered_images
 
-    def save_csv_with_derotation_data(self):
-        """Saves a csv file with the rotation angles by line and frame,
-        and the rotation on signal.
-        It is saved in the saving folder specified in the config file.
+    def find_center_of_rotation(self) -> Tuple[int, int]:
+        """Find the center of rotation by fitting an ellipse to the largest
+        blob centers.
+
+        Step 1: Calculate the mean images for each rotation increment.
+        Step 2: Find the blobs in the mean images.
+        Step 3: Fit an ellipse to the largest blob centers and get its center.
+
+        Returns
+        -------
+        Tuple[int, int]
+            The coordinates center of rotation (x, y).
         """
-        df = pd.DataFrame(
-            columns=[
-                "frame",
-                "rotation_angle",
-                "clock",
-            ]
+        logging.info(
+            "Fitting an ellipse to the largest blob centers "
+            + "to find the center of rotation..."
         )
+        mean_images = self.calculate_mean_images(self.image_stack)
 
-        df["frame"] = np.arange(self.num_frames)
-        df["rotation_angle"] = self.rot_deg_frame[: self.num_frames]
-        df["clock"] = self.frame_start[: self.num_frames]
+        logging.info("Finding blobs...")
+        coord_first_blob_of_every_image = self.get_coords_of_largest_blob(
```

Comment:
Yes definitely that is possible. I have made a better version of this algorithm in another branch... it will come soon. [Sneak peek](https://github.com/neuroinformatics-unit/derotation/blob/feature/ease-batch-processing/derotation/analysis/metrics.py)